Tasks/SonarQubePostTest/SonarQubeBuildBreaker.ps1

```diff
@@ -35,8 +35,8 @@ function FailBuildOnQualityGateStatus
     {        
         $dashboardUrl = GetTaskContextVariable "MSBuild.SonarQube.ProjectUri"
         
-        Write-Host "##vso[task.logissue type=error]The SonarQube quality gate associated with this build has failed. For more details see $dashboardUrl"
-        Write-Host "##vso[task.complete result=Failed;]"
+        Write-VstsTaskError "The SonarQube quality gate associated with this build has failed. For more details see $dashboardUrl"
```

Comment:
Here too, you can simply Write-Error or throw. Unless you are changing the default error action preference [set by the task lib](https://github.com/Microsoft/vsts-task-lib/blob/master/powershell/Docs/ErrorsWarningsAndTaskResult.md) (Stop), then you don't need to manage setting the result explicitly.